spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/session/SessionStoreType.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2012-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.autoconfigure.session;
+
+/**
+ * Supported Spring Session data store types.
+ *
+ * @author Tommy Ludwig
+ * @since 1.4.0
+ */
+public enum SessionStoreType {
+
+	/**
+	 * Redis backed sessions.
+	 */
+	REDIS,
+
+	/**
+	 * Hazelcast backed sessions.
+	 */
+	HAZELCAST,
+
+	/**
+	 * Simple in-memory map of sessions.
+	 */
+	SIMPLE,
```

Comment:
@snicoll 

>  you lost me with the SLIGHTLY_TRICKY part to be honest. Is that ironic?

Sorry about that. I was trying to point out that other stores do not use their complexity to determine the annotation. Instead the name reflects the actual data store used. When I use `SIMPLE` this is *subjective* adjective of the data store and does not clearly define the store.

There are likely multiple ways to implement a simple data store. I'd like to ensure that the enum describes exactly what is being used so there is no confusion. In this instance, `HASH_MAP` describes this well. Perhaps it could be even better defined as `CONCURRENT_HASH_MAP`.